include/boost/polygon/detail/voronoi_predicates.hpp

```diff
@@ -1129,6 +1129,27 @@ class voronoi_predicates {
             site1, site2, site3, segment_index, c_event,
             recompute_c_x, recompute_c_y, recompute_lower_x);
       }
+      // All points needs to be unique or the dot product calculation will not be valid
+      bool unique_endpoints = !(site1.point0() == site2.point0() ||
```

Comment:
Do we actually actually need this now that you are not calling `ppp` anymore? 
i.e. don't we only want that `site3.is_segment() == true`? (which should be required for calling `pps` anyway)

Reply:
without a unique endpoint test the diagram generation will fail on examples like 
```
0
3
-5138 -5149 -5038 -5142
-5042 -5069 -5165 -5162
-5011 -5195 -5404 -5134
```
or 
```
0
2
-5093 -5402 -5362 -5008
-5200 -5273 -5273 -5265
```
I'm not exactly sure why, but I guess it ties in to your follow up question:

> That can lead to missing voronoi vertices right?

Yes